### PR TITLE
Project integration changes

### PR DESCRIPTION
**Pull request by georgebrock**
A grab-bag of hacky changes that I made to get ActionClient working with a real world project. Each commit works around a problem that I encountered, and was driven out by failures in the host project's test suite rather than the ActionClient test suite.

This PR should not be merged, I'm opening it only as an artefact to link to in a Basecamp discussion.

If any of the commits here are determined to be good ideas, they probably warrant their own PRs with test coverage and more thorough documentation.

### CODE REVIEW

File: lib/action_client/middleware/response_parser.rb [hunk 1]
```diff
@@ -12,7 +12,7 @@ def call(env)
 
         if body.present?
           if content_type.starts_with?("application/json")
-            body = JSON.parse(body)
+            body = JSON.parse(body, object_class: HashWithIndifferentAccess)
```
Comment (seanpdoyle) [inline lib/action_client/middleware/response_parser.rb:15]: 🤯

Comment (mike-burns) [reply lib/action_client/middleware/response_parser.rb:15]: How about `JSON.parse(body, symbolize_keys: true)` instead? I've never seen a case where I wanted a string.

Comment (georgebrock) [reply lib/action_client/middleware/response_parser.rb:15]: I'd be fine with `symbolize_keys`.

I went with `HashWithIndifferentAccess` so I could not change the code in my client project that expected symbols and also not change any code elsewhere in the library that expected strings.

Comment (seanpdoyle) [reply lib/action_client/middleware/response_parser.rb:15]: According to the docs, the option name is `symbolize_names`: https://ruby-doc.org/stdlib-2.6.3/libdoc/json/rdoc/JSON.html#method-i-parse

Weird!

**Comment by seanpdoyle** [comment]
Thanks @georgebrock !

I've re-structured the project's history, including a commit that steals some of the changes from this PR: https://github.com/seanpdoyle/action_client/commit/4cc28169e1c5290059cf87810b9fae0833771342

I omitted the `HashWithIndifferentAccess` commit, since that feels like a consumer-level concern. Hopefully, we can make it easy to do so!

**Pull request closed without merging**